core/primer.md

```diff
@@ -33,7 +34,130 @@ associating those declarations with application endpoints. As a result, the
+endpoints to evolve. xRegistry allows indicating a registry’s compatibility
+strategy which is enforced when making changes to the registry, avoiding
+mistakes. The specifications further also enables marking endpoints as
+deprecated, possibly providing an alternative endpoint, making these easier to
+discover for consumers. Finally, event schemas are also version aware, storing
+multiple versions of an event’s data schema.
+
+### Schema validation
+
+The registry contains granular definitions for events defining their schemas,
+which serve as a contract between publishers and consumers. This doesn’t only
+apply to the message payload, but also to the message metadata, allowing
+granular control of required fields on the metadata level. The registry does not
+only serve as a definition of the required schemas, but can also be used to
+enforce it. A registry service could use the registry to perform validation
+on-publish, rejecting any events that don’t adhere to the schema for that
+context, avoiding the problem of poison messages.
+
+### Payload reduction
+
+Schema information enables real-time analysis of incoming data based on its
+contextual meaning, which is why applications often use serialization mechanisms
+which include schema information (such as JSON) in payloads. Having a
+centralized registry, enables applications to rely on lighter weight
+serialization formats, significantly reducing bandwidth, while still allowing
+accurate real-time analysis of incoming data using the registry.
+
+### Schema-based data contract generation
+
+xRegistry provides a specification to manage metadata in files (for example,
+stored next to relevant source code), but the metadata can also be managed in a
+Registry service, assuming it adheres to the xRegistry API specification.
+
+This enables teams to use code generation to generate relevant data contracts
+when interacting with these endpoints. This removes the need for shared code
+packages and avoids any bugs on the data contract level.
+
+### Producer and consumer generation
+
+Further, building upon the idea of schema-based data contract generation, even
+client-code can be generated. When the schema is defined in a vendor-agnostic,
+well-defined format, tooling can be built upon this foundation to generate
+broker-specific consumers that can consume events from endpoints defined in the
+registry. This is not only limited to event consumers, but may also include
+generation of event store inserts. This can further speed up and streamline the
+development of both producers and consumers, significantly reducing development
+and testing time and minimizing the risk of introducing bugs at this level.
+Clemens Vasters built a proof-of-concept that shows this approach in
+the [Avrotize VS Code Extension](https://github.com/clemensv/avrotize).
+
+### Basis for further developments
+
+CloudEvents serve as a well-defined event-format, which is applicable
```

Comment:
I get nervous when people call CE an "event-format" because it's not really meant to be that, even though it does offer a JSON event format that people MAY use if they don't already have a format defined. But, and this is critical, CE is really more about its metadata and where the metadata can be found independent of the event format. CE is NOT meant to be "one event format to rule them all". That has been tried and failed, many times.

So, perhaps instead say something like: CloudEvents serve as a mechanism to aid in the delivery of events from a producer to a consumer, which is applicable independently ....

Reply:
Addressed in [`08032ac` (#219)](https://github.com/xregistry/spec/pull/219/commits/08032accc895eb10afd75e1bd3e85793f93d18b5)